vault/logical_system.go

```diff
@@ -692,6 +693,42 @@ func NewSystemBackend(core *Core, config *logical.BackendConfig) (logical.Backen
 				HelpSynopsis:    strings.TrimSpace(sysHelp["audited-headers"][0]),
 				HelpDescription: strings.TrimSpace(sysHelp["audited-headers"][1]),
 			},
+			&framework.Path{
+				Pattern: "plugin-catalog/$",
+
+				Fields: map[string]*framework.FieldSchema{},
+
+				Callbacks: map[logical.Operation]framework.OperationFunc{
+					logical.ListOperation: b.handlePluginCatalogList,
+				},
+
+				HelpSynopsis:    strings.TrimSpace(sysHelp["plugin-catalog"][0]),
+				HelpDescription: strings.TrimSpace(sysHelp["plugin-catalog"][1]),
+			},
+			&framework.Path{
+				Pattern: "plugin-catalog/(?P<name>.+)",
+
+				Fields: map[string]*framework.FieldSchema{
+					"name": &framework.FieldSchema{
+						Type: framework.TypeString,
+					},
+					"sha_256": &framework.FieldSchema{
+						Type: framework.TypeString,
+					},
+					"command": &framework.FieldSchema{
+						Type: framework.TypeString,
```

Comment:
There's no built-in description here (there probably should be) and as a result it's unclear from this file/endpoint whether what's here is a full command, including arguments, that will be executed, or whether it's simply a path to a file. Based on discussions I think it's the latter but sounds like the former.

If the former is desired (maybe as an alternative) we could support a command argument that takes in a slice. However, if we end up switching to a goroutine based go-plugins method before release this will *likely* not be necessary as then you wouldn't even need that support to reinvoke Vault with `plugin-exec`.